### PR TITLE
docs: Enhance web communication and localization

### LINKED ISSUE #257: docs: Improve web communication and documentation
I’ve noticed that some repositories have localized README.md files, which could be useful for non-English speakers. Even though the majority of GitHub users speak English more or less fluently, it might still be helpful.

I came across this simple example (https://github.com/jonatasemidio/multilanguage-readme-pattern) that looks quite interesting. I can take care of this new feature if the development team is okay with it (fork + PR for the pattern in main README.md, then ShlomoCode could translate in Hebrew if they want).

### LINKED ISSUE #272: docs: Update README.md
Following #257, I will update README.md. This will be addressed soon in #258.

### LINKED ISSUE #285: [BUG] Modifier keys toggle previewed applications
## Describe the bug
When hovering an application with opened windows, pressing any modifier key (Shift, Command, Control, Fn or Option) will automatically toggle the first window. This might interfere with app keyboard shortcuts.

## Steps to reproduce
- Hover over an application icon with opened windows.
- Press Shift, Command, Control, Fn or Option.

## Expected behavior
Modifier keys like that should not bring the window to front.

## Environment
 - macOS version: macOS 14.5
 - App version: 1.1.6

- [x] I have reviewed the open issues to ensure this is not a duplicate

### PR DESCRIPTION
**Pull request by illavoluntas**
## Describe your changes
This PR creates a brand new HTML frontpage for the website https://dockdoor.net and localizes it (using collaborative translations, Crowdin).

## Related issue number(s) and link(s)
Closes #257
Closes #272
Closes #285. 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have followed conventional commit guidelines

## Core Functionality Changes
- This PR **does not** change core files.

## Checklist before merging this PR
- [ ] Mark this PR **ready for review**

### CODE REVIEW

**Comment by illavoluntas** [comment]
Tasks for myself:

- [ ] Add scrolling window title loop
- [ ] Add window title screenshot
- [ ] WIP.

**Comment by illavoluntas** [comment]
**Oncoming minor tasks**

- [ ] Correct line height for quick links
- [ ] Correct line height for download buttons

**Comment by illavoluntas** [comment]
@ejbills, @ShlomoCode
Wow, this PR is at last ready for merging! Do not hesitate if you have any suggestions or if you want to share your opinion. I'm open to all advices!

**Comment by illavoluntas** [comment]
@ejbills, @ShlomoCode
Wow, this PR is at last ready for merging! Do not hesitate if you have any suggestions or if you want to share your opinion. I'm open to all advices!

**Comment by illavoluntas** [comment]
I recommend squash and merge because that's a lot of commits here.

**Comment by ejbills** [comment]
Do the translations tie into Crowdin?

**Comment by illavoluntas** [comment]
> Do the translations tie into Crowdin?

I’m afraid to break Localizable.xcstrings by integrating translation.json to crowdin.yml.

**Comment by illavoluntas** [comment]
Well, let me try

File: README.md [hunk 1]
```diff
@@ -1,92 +1,145 @@
...
-3. Open the project in Xcode.
-4. Build and run the project.
+## Frequently Asked Questions
 
-## How to Contribute
+### How to access DockDoor settings?
+If you disabled `Show Menu Bar Icon` in `General`, search for `DockDoor` using the built-in Spotlight or another searching app. Open DockDoor and the settings should open automatically.
 
-Contributions to DockDoor are welcome! Here’s how you can get started:
+If this option is enabled, click on *door* icon on the menu bar and select `Open Settings` in the contextual menu.
```
Comment (ejbills) [inline README.md:80]: should be:

click on the *door* icon in the menu bar

File: README.md [hunk 2]
```diff
@@ -1,92 +1,145 @@
...
+
+### The application I want to quit does not respond.
+You can either wait for the application to become responsive again, or force quit it. To do this, hold **`Option`** (⌥) while clicking the `Quit` button.
+
+### Why is DockDoor only for macOS 13 and newer?
```
Comment (ejbills) [inline README.md:85]: I don't think we need this question in the README

Comment (illavoluntas) [reply README.md:85]: What question? The extract you preview shows 2 questions.

**Comment by ejbills** [comment]
> > Do the translations tie into Crowdin?
> 
> I’m afraid to break Localizable.xcstrings by integrating translation.json to crowdin.yml.

I see now you created a new crowdin page. That is a good approach, but I would like to own the crowdin so it can all be under the one official account. What would I need to replace from your PR?

File: README.md (hunk 2 above)
Comment (ejbills) [reply README.md:85]: This one: Why is DockDoor only for macOS 13 and newer

File: README.md (hunk 1 above)
Comment (illavoluntas) [reply README.md]: Resolved in 7102ac8585dc9d363092754a4d74b9955e767d7d

**Comment by illavoluntas** [comment]
> > > Do the translations tie into Crowdin?
> > 
> > 
> > I’m afraid to break Localizable.xcstrings by integrating translation.json to crowdin.yml.
> 
> I see now you created a new crowdin page. That is a good approach, but I would like to own the crowdin so it can all be under the one official account. What would I need to replace from your PR?

I can transfer ownership to you on Crowdin.

File: README.md (hunk 2 above)
Comment (illavoluntas) [reply README.md:85]: @ejbills Is it `The application I want to quit does not respond.`?

Comment (illavoluntas) [reply README.md]: Resolved in 7df39ebcf641aa5e12cae96a080472255b9cb2d2

**Comment by illavoluntas** [comment]
Oh, you can set a separate crowdin.yml. I'm trying to integrate Crowdin automation for testing purposes.

**Comment by illavoluntas** [comment]
@ejbills, I can't find a way of having both Localizable.xcstrings and translation.json in the same repository without getting conflicts in Crowdin... Even though I set up Crowdin with a custom yml file, it still wants to translate Localizable.xcstrings...

**Comment by illavoluntas** [comment]
Best thing I can suggest here is creating a dedicated repository for the website. This way you can have a clean workspace for Crowdin.

**Comment by ShlomoCode** [comment]
1. It seems that the site is in RTL view even though the Hebrew language does not exist (pay attention to the position of the dots)
CleanShot 2024-08-31 at 20 47 10@2x
2. It is not clear what happens when the button is clicked, I think the icon should be as the situation that will happen after clicking it
CleanShot 2024-08-31 at 20 47 43@2x
3. The dark blue color is too dominant, it is not DockDoor's brand color and I think it should be reduced

**Comment by illavoluntas** [comment]
> 1. It seems that the site is in RTL view even though the Hebrew language does not exist (pay attention to the position of the dots)
>    CleanShot 2024-08-31 at 20 47 10@2x
> 2. It is not clear what happens when the button is clicked, I think the icon should be as the situation that will happen after clicking it
>    CleanShot 2024-08-31 at 20 47 43@2x
> 3. The dark blue color is too dominant, it is not DockDoor's brand color and I think it should be reduced

1. Language detection is automatic. If you have a Hebrew system language, it will show this. This is temporary as I am waiting for Hebrew translations.
2. Agreed, but do you mean only sun / or moon at the beginning?
3. What would you choose as a dominant color? DockDoor does not have a brand color; choosing pink or light blue is in my opinion not very suitable for web display (not much contrast with white).

**Comment by illavoluntas** [comment]
1. Visit https://illavoluntas.github.io/?lang=en for direct English language.

**Comment by illavoluntas** [comment]
Remark 2 was fixed in c930148e3b36182ee42fb08a6363f4d330ecb403.

**Comment by ShlomoCode** [comment]
> (https://github.com/ejbills/DockDoor/commit/9c989d716e3714345a07bee21f76a4f0e16cd297). Is it what you meant?

Yes!

**Comment by illavoluntas** [comment]
What do you suggest as a dominant color?

**Comment by ShlomoCode** [comment]
> 3. What would you choose as a dominant color? DockDoor does not have a brand color; choosing pink or light blue is in my opinion not very suitable for web display (not much contrast with white).

Why should the text be white? I would do it that way
CleanShot 2024-08-31 at 21 27 24@2x
CleanShot 2024-08-31 at 21 33 35@2x

**Comment by illavoluntas** [comment]
What does @ejbills think about it?

**Comment by illavoluntas** [comment]
Oh, one thing left. Adding DOCS.md (I already made screenshots in https://github.com/illavoluntas/illavoluntas.github.io/tree/main/resources/settings) with a tiny documentation (adding #285 too). After I’m done writing it, I think you should add a question mark button adapted to each tab (maybe not Updates). See https://github.com/ejbills/DockDoor/issues/257#issuecomment-2308510302

**Comment by illavoluntas** [comment]
> Best thing I can suggest here is creating a dedicated repository for the website. This way you can have a clean workspace for Crowdin.

@ejbills, I think you will have to do that. Plus creating a Crowdin project.

**Comment by ejbills** [comment]
> > Best thing I can suggest here is creating a dedicated repository for the website. This way you can have a clean workspace for Crowdin.
> 
> 
> 
> @ejbills, I think you will have to do that. Plus creating a Crowdin project. This way it also separates the website and the app, and source code archives (when you publish a new release) will still weigh less than 5 MB instead of 20 MB with all these files I added for this PR. 

The site files wouldn't be included in the compiled app. 

I'm a bit stuck, if we replace it with a new repo just for the site, I wouldn't be able to use the domain on this repo. Which means, the sparkle updater would be broken as I already switched the domain for this repo. If I switch it back, I'd have to make dockdoor.net forward to dockdoor.kepler.cafe for updates to work, consequentially meaning I wouldn't be able to use dockdoor.net on the other repo.

**Comment by illavoluntas** [comment]
I know the site wouldn’t be included to the final DMG, I meant the source code ZIP and TAR GZ. Contributors would have to get site files when they clone the repo. 

As for the updater, I understand your problem. Maybe I will find a way to make Crowdin work **only** for translation.json (in the website Crowdin project) by modifying parameters in my test project. I tried linking to another file than crowdin.yml, customized, but it failed.

**Comment by ShlomoCode** [comment]
Maybe we can use a crowdin alternative for the site. I have listed a few in https://github.com/ejbills/DockDoor/issues/114.

**Comment by ejbills** [comment]
Nothing I have seen in the crowdin docs to specify we can't use one, shared crowdin.yml.

See: https://developer.crowdin.com/configuration-file/#ignoring-files-and-directories

We just need to specify a separate project id. However, this is locked into the paid tier of Crowdin. See: https://community.crowdin.com/t/how-to-add-2-different-integrations-for-the-same-project/3289/2

I need to apply for a free open source license.

**Comment by illavoluntas** [comment]
> I need to apply for a free open source license.

What do you mean?

**Comment by illavoluntas** [comment]
Well, you might want to try.

**Comment by ejbills** [comment]
I was approved for the open-source license, I need to look into how we can integrate this into the crowdin account. I will get back to you.

**Comment by illavoluntas** [comment]
> I was approved for the open-source license, I need to look into how we can integrate this into the crowdin account. I will get back to you.

Nice! Thank you for making the demand!

**Comment by illavoluntas** [comment]
I think the main thing to modify here is `crowdin.yml`. If you succeed, will it be necessary to create a separate Crowdin project for the website?

**Comment by illavoluntas** [comment]
@ejbills Any update about it?

**Comment by ejbills** [comment]
> @ejbills Hi! Any update about it?

It's on my roster, I am a bit busy atm. I have this on my mind I promise, I will get to this in the coming days.

**Comment by illavoluntas** [comment]
> > @ejbills Hi! Any update about it?
> 
> It's on my roster, I am a bit busy atm. I have this on my mind I promise, I will get to this in the coming days.

No problem! You don’t have to hurry at all. I’m very busy too and I’m essentially available on weekends… which makes the workflow slower…

**Comment by illavoluntas** [comment]
Well, there were some little adjustments left to be made…But they are minor, so I’ll review them and make a new PR :wink:

**Comment by ejbills** [comment]
@illavoluntas merged! Thanks so much for this wonderful PR. I realize the changes I made recently regarding the default keybind and related options will need to be reflected in the README and FAQ sections (now defaults to option tab). I will do this tomorrow if I have some time.

**Comment by illavoluntas** [comment]
You’re welcome! What about Crowdin? The old link (my project) is still on README…

**Comment by illavoluntas** [comment]
Also, could my work be please mentioned on the next release? 
You finally may want to add the website link to the description field on https://github.com/ejbills/DockDoor. See https://stackoverflow.com/questions/67217281/adding-github-pages-url-to-github-repositorys-about-section

**Comment by ejbills** [comment]
> Also, could my work be please mentioned on the next release? 

Of course 😁

I fixed the readme crowdin link and added the github pages link to the sidebar

**Comment by illavoluntas** [comment]
@ejbills Could you remove the `/` after the website link?

**Pull request merged**